ci/disallow_c_comments_in_migrations.sh

```diff
@@ -12,17 +12,17 @@ source ci/ci_helpers.sh
 # and reusing them if needed. GNU sed unfortunately does not support lookaround assertions.
 
 # /* -> --
-find src/backend/distributed/sql/*.sql -print0 | xargs -0 sed -i 's#/\*#--#g'
+find src/backend/distributed/sql/{,**/}*.sql -print0 | xargs -0 sed -i 's#/\*#--#g'
```

Comment:
I believe this still doesn't cover if we have sql files in 2 deeper levels from `sql` such as `src/backend/distributed/sql/downgrades/newFolder/xxx.sql`. Or folders inside `udfs` folder.

It would be better to not have any limits on how deep we can go, what do you think?

Reply:
`**` covers all directories down the line. You can check this out for yourself.

Create test directories and test files.

	sh$ mkdir -p globtesting/a/b/c/d
	sh$ find globtesting | xargs -n1 -I {} touch "{}/dummy.sql"

Check the file tree

	  sh$ tree glob-testing
	  glob-testing
	  |-- a
	  |   |-- b
	  |   |   |-- c
	  |   |   |   |-- d
	  |   |   |   |   `-- dummy.sql
	  |   |   |   `-- dummy.sql
	  |   |   `-- dummy.sql
	  |   `-- dummy.sql
	  `-- dummy.sql
	  
	  4 directories, 5 files

Test find and `**` behaviour

	  sh$ find globtesting/{,**/}*.sql
	  globtesting/dummy.sql
	  globtesting/a/b/c/d/dummy.sql
	  globtesting/a/b/c/dummy.sql
	  globtesting/a/b/dummy.sql
	  globtesting/a/dummy.sql
	  globtesting/dummy.sql

Reply:
The only problem is the repeating 

	globtesting/dummy.sql

line. Maybe I can use 

	sh$ find globtesting/**/*.sql

instead of 

	sh$ find globtesting/{,**/}*.sql

Reply:
umm, I actually tried this before writing:

```bash
find src/backend/distributed/sql/{,**/}*.sql -print
...
src/backend/distributed/sql/downgrades/citus--10.0-4--9.5-1.sql
src/backend/distributed/sql/downgrades/citus--10.1-1--10.0-4.sql
src/backend/distributed/sql/downgrades/citus--10.2-1--10.1-1.sql
src/backend/distributed/sql/downgrades/citus--9.2-4--9.2-2.sql
src/backend/distributed/sql/downgrades/citus--9.3-2--9.2-4.sql
src/backend/distributed/sql/downgrades/citus--9.4-1--9.3-2.sql
src/backend/distributed/sql/downgrades/citus--9.5-1--9.4-1.sql
```

However if I write:

```bash
talha@DESKTOP-EJEU49S:~/citus$ find src/backend/distributed/sql/downgrades/{,**/}*.sql -print
src/backend/distributed/sql/downgrades/citus--10.0-4--9.5-1.sql
src/backend/distributed/sql/downgrades/citus--10.1-1--10.0-4.sql
src/backend/distributed/sql/downgrades/citus--10.2-1--10.1-1.sql
src/backend/distributed/sql/downgrades/citus--9.2-4--9.2-2.sql
src/backend/distributed/sql/downgrades/citus--9.3-2--9.2-4.sql
src/backend/distributed/sql/downgrades/citus--9.4-1--9.3-2.sql
src/backend/distributed/sql/downgrades/citus--9.5-1--9.4-1.sql
**src/backend/distributed/sql/downgrades/x/citus--9.5-1--9.4-1.sql**
```

I created the `x` folder to try this and you can see that it is not shown in the output for the first command. What am I missing?

Reply:
Your bash has a different option set.

```
globstar
    If set, the pattern ** used in a pathname expansion context will
    match all files and zero or more directories and subdirectories.
    If the pattern is followed by a /, only directories and
    subdirectories match.
```

I added `shopt -s globstar` that enables this feature.

Reply:
After this change, the script ended up fixing more sql files than before.